pkg/api/v1alpha1/cluster_webhook_test.go

```diff
@@ -765,7 +765,7 @@ func TestClusterValidateUpdateAWSIamNameImmutableEmpty(t *testing.T) {
 	c.Spec.IdentityProviderRefs = []v1alpha1.Ref{}
 
 	g := NewWithT(t)
-	g.Expect(c.ValidateUpdate(context.TODO(), cOld, c)).Error().To(MatchError(ContainSubstring("spec.identityProviderRefs.AWSIamConfig: Forbidden: field is immutable")))
+	g.Expect(c.ValidateUpdate(context.TODO(), cOld, c)).Error().To(Succeed())
```

Comment:
Should we rename these tests? Or change what we are testing. Like it reads like we want to test immutability.

Reply:
Updated now